### DISCUSSION TITLE
Nested repeating group in repeater

### DISCUSSION THREAD

**Discussion by zabatonni** [directus/directus#15629] [Archive]
Is it possible to nest repeater with groups? Idea is to create "website builder". So you would have "content" field where you can select from pre-defined blocks (ie. hero, features, simple-text, ...) and each of these blocks would have repeating groups as well ie. hero would have group slide with fields title, description, image. And user would be able to place as many hero sections with as many slides in each as needed.

Output should be JSON for example `
"blocks":[{"type":"hero","slides":[{"title":"Lorem Ipsum","description":"Lorem..."],[{...}]},{"type":"features"}]`

**Comment by lessborders** [comment]
You have the same idea as me. Repeaters are cool but very lacking. It would be cool if we could create our own components, and drag and drop them into a "builder" or even a "grid repeater" field.

**Reply by benhaynes** [reply]
I like this idea a lot, but it's pretty vague. Maybe we could figure out a set of 80/20 capabilities for a full-featured and aesthetically pleasing block editor?

**Reply by lessborders** [reply]
If you wanted to get very advanced, you could make something like WPBakery Page Builder for WordPress.
But it would also be enough if we could just create some kind of "Grid" to put our repeater fields, or M2A relationships in.

**Reply by lessborders** [reply]
Option 2 would be to let us create Variable Repeater Fields with no predefined template - so each item in a repeater could have a different schema. Because right now the only way to dynamically mount components is like this:
![image](https://user-images.githubusercontent.com/100006598/190882600-77ddc6a3-1b9c-4ee3-aed5-68dc2c16d077.png)

**Reply by zabatonni** [reply]
> I like this idea a lot, but it's pretty vague. Maybe we could figure out a set of 80/20 capabilities for a full-featured and aesthetically pleasing block editor?

I would not lock this featured to be used just for "block editor". It can be used to produce ld+json schema or whatever needs to be delivered into app/website.
